rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java

```diff
@@ -67,22 +92,38 @@ private void performPostScanActions(Bundle bundle) {
         try {
             displayStatusMessageAndSymbol(response.sensorTriggered, response.expirationDate);
             populateBarcodeDetails();
-        } catch (ParseException e) {
+        } catch (JSONException | ParseException e) {
             Timber.e(e);
         }
     }
 
-    private void populateBarcodeDetails() {
-        setBarcodeResult(R.id.barcode_product_id, response.productId);
-        setBarcodeResult(R.id.barcode_serial_no, response.serialNumber);
-        setBarcodeResult(R.id.barcode_additional_id, response.additionalIdentifier);
-        setBarcodeResult(R.id.barcode_lot_no, response.lot);
-        setBarcodeResult(R.id.barcode_expiration_date, response.expirationDate);
-        setBarcodeResult(R.id.barcode_is_sensor_triggered, response.sensorTriggered ? getString(R.string.yes) : getString(R.string.no));
+    private void populateBarcodeDetails() throws JSONException {
+
+        if (enableBatchScan) {
+            JSONObject scanObject = new JSONObject();
+            scanObject.put("productId", response.productId);
+            scanObject.put("serialNumber", response.serialNumber);
+            scanObject.put("additionalIdentifier", response.additionalIdentifier);
+            scanObject.put("lot", response.lot);
+            scanObject.put("expirationDate", response.expirationDate);
+            dataArray.put(scanObject);
+            viewBinding = DataBindingUtil.inflate(getLayoutInflater(), R.layout.item_one_scan, parentView, false);
+        }
+
+        setBarcodeResult(viewBinding, R.id.barcode_product_id, response.productId);
+        setBarcodeResult(viewBinding, R.id.barcode_serial_no, response.serialNumber);
+        setBarcodeResult(viewBinding, R.id.barcode_additional_id, response.additionalIdentifier);
+        setBarcodeResult(viewBinding, R.id.barcode_lot_no, response.lot);
+        setBarcodeResult(viewBinding, R.id.barcode_expiration_date, response.expirationDate);
+        setBarcodeResult(viewBinding, R.id.barcode_is_sensor_triggered, response.sensorTriggered ? getString(R.string.yes) : getString(R.string.no));
+
+        if (enableBatchScan) {
+            parentView.addView(viewBinding.getRoot());
```

Comment:
Any reason not to have this inside this if block https://github.com/onaio/rdt-standard/pull/592/files#diff-dd323e705143722471da28e8ee65c1b15a10c71d31bc603b624d0f5ee1f67ea7R102?

Reply:
Since we only need this for `enableBatchScan`.

Reply:
yea, I meant could it move into this if block https://github.com/onaio/rdt-standard/pull/592/files#diff-dd323e705143722471da28e8ee65c1b15a10c71d31bc603b624d0f5ee1f67ea7R102?

Reply:
fixed.